qiita_pet/templates/analysis_selected.html

```diff
@@ -1,8 +1,15 @@
 {% extends sitebase.html %}
 {% block head %}
 <script type="text/javascript">
+var ws = new WebSocket('ws://' + window.location.host + '/analysis/selected/socket/');
+ws.onmessage = function(evt) { if(evt.data != 'true') {ws.close();} };
```

Comment:
So it will close if there is an error raised during the action in the socket, but not if something goes wrong silently. This will take care of those silent issues.